File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    void bfs(int cur, int n, int[][] computers, boolean visited[]){
        Queue<Integer> queue = new LinkedList<>();
        queue.add(cur);
        
        while(!queue.isEmpty()){        
            int node = queue.poll();
            visited[node] = true;

            for(int i = 0; i < n; i++){
                if(computers[node][i] == 1 && i != node && !visited[i]){
                    queue.add(i);
                }
            }
        }
    }
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for(int i = 0; i < n; i++){
            visited[i] = false;
        }
        
        for(int i = 0; i < n; i++){
            if(!visited[i]){
                answer++;
                bfs(i, n, computers, visited);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    n  = len(computers)
    visited = [False] * n
    
    def dfs(cur, computers):
        dq = deque()
        dq.append(cur)
        
        while dq:
            node = dq.popleft()
            visited[node] = True
            
            for i in range(n):
                if computers[node][i] and not visited[i]:
                    dq.append(i)
    
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(i, computers)
            
    
    return answer